nemo/deploy/query.py

```diff
@@ -0,0 +1,229 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from abc import ABC, abstractmethod
+
+import numpy as np
+
+from .utils import str_list2numpy
+
+use_pytriton = True
+try:
+    from pytriton.client import DecoupledModelClient, ModelClient
+except:
+    use_pytriton = False
+
+
+class NemoQueryBase(ABC):
+    def __init__(self, url, model_name):
+        self.url = url
+        self.model_name = model_name
+
+    @abstractmethod
+    def query_llm(
+        self,
+        prompts,
+        stop_words_list=None,
+        bad_words_list=None,
+        no_repeat_ngram_size=None,
+        max_output_token=512,
+        top_k=1,
+        top_p=0.0,
+        temperature=1.0,
+        random_seed=None,
+        task_id=None,
+        lora_uids=None,
+        init_timeout=60.0,
+    ):
+        pass
+
+
+class NemoQuery(NemoQueryBase):
```

Comment:
Please denote this as NeMoLLMPytritonQuery (or something that properly denotes it is an wrapper for querying a pytriton based llm deployment)

Reply:
Changed it to NemoQueryLLM since it can be used to query any server. Just used pytriton classes to send query for convenience. So, we don't need to use PyTriton in the name.